src/load.lisp

```diff
@@ -0,0 +1,335 @@
+;;;                        {
+;;;                            "name": "jso",
+;;;                            "main": "jso.js"
+;;;                        }
+;;;          So, (require "./lib/app/jso/jso") will load file 'jso.js' from './lib/app/jso'
+;;;
+;;;          See node.js documentation for 'require' function
+;;;
+;;; hook  - #() store place for js-code
+;;;
+;;;          (setq bin #())
+;;;          (load "file1.lisp" :hook bin)
+;;;          (load "file2.lisp" :hook bin)
+;;;          (load "file3.lisp" :hook bin :output "lib.js")
+;;;              => will be bundle js-code file1, file2, file3 from bin to "lib.js"
+;;;
+;;;         you will be use (requie "./lib") or use html tag:
+;;;         <script src="lib.js" type="text/javascript" charset="utf-8"></script>
+;;;         without compilation.
+;;;
+;;;
+;;; have a fun
+;;;
+
+(defun node-environment-p ()
+  (if (find :node *features*) t))
+
+
+(defun load (name &key verbose (sync (node-environment-p)) output place hook)
+  (terpri)
+  (cond
+    ;; node.js/electron platform
+    ((node-environment-p)
+     (if place
+         (setq place (concat place "./"))
+         (setq place "./"))
+     (if sync
+         (loader-sync-mode name verbose output place hook)
+         (loader-async-mode name verbose output place hook)))
+    ;; browser platform
+    (t
+     (when sync
+       (warn "sync mode only for node/electron platform~%will be used async mode"))
+     (when (or output bin)
+       (warn "output/hook options only for node/electron platform"))
+     (loader-browser-mode name verbose)))
+  (values))
+
+
+(defun loader-browser-mode (name verbose)
+  (_xhr_receiver_
+   name
+   (lambda (input)
+     (_load_form_eval_ (_ldr_ctrl-r_replace_ input) verbose))
+   (lambda (url status)
+     (format t "~%Load: Can't load ~a~%     Status ~%" url status))))
+
+;;; alowe make bundle from source received from local fs (by FILE:)
+;;; or from remote resourse (by HTTP:)
+(defun loader-async-mode (name verbose bundle-name place hook)
+  (_xhr_receiver_
+   name
+   (lambda (input)
+     (_load_eval_bundle_ (_ldr_ctrl-r_replace_ input) verbose bundle-name place hook))
+   (lambda (url status)
+     (format t "~%Load: Can't load ~a~%     Status ~%" url status))))
+
+;;; sync mode
+(defun loader-sync-mode (name verbose bundle-name place hook)
+  ;; check what input file exists
+  (unless (#j:Fs:existsSync name)
+    (error "No such file ~s" name))
+  (_load_eval_bundle_
+   (_ldr_ctrl-r_replace_ (#j:Fs:readFileSync name "utf-8"))
+   verbose
+   bundle-name
+   place
+   hook))
+
+(defun _load_eval_bundle_ (input verbose bundle-name place hook)
+  (let (stream eof code expr rc code-stor fbundle)
+    (setq eof (gensym "LOADER"))
+    (if hook (setq code-stor hook))
+    (when bundle-name
+      (if hook
+          (setq code-stor hook)
+          (setq code-stor #()))
+      (setq fbundle t))
+    (setq stream (make-string-input-stream input))
+    (tagbody sync-loader-rdr
```

Comment:
I think I already mentioned, but `tagbody` is rather inefficient in JSCL. It's better to use other looping constructs. But it's okay.

Reply:
Not sure, but see it.
Thank.

Reply:
`do`/`dolist`/`dotimes` also implemented through `tagbody`

Reply:
They are not actually 

https://github.com/jscl-project/jscl/blob/master/src/boot.lisp#L171

they are implemented in terms of `%while`, which is an internal primitive that compiles to a `while` loop in JS.